File: 03_REVC_reverse_complement/haskell/03_reverse_complement.hs
module Main where

main :: IO()
main = do
  seq <- getLine
  putStrLn $ reverse $ map complement seq

complement :: Char -> Char
complement 'A' = 'T'
complement 'C' = 'G'
complement 'G' = 'C'
complement 'T' = 'A'
complement  _  = 'N' 

File: 03_REVC_reverse_complement/python/03_reverse_complement.py
#!/usr/bin/env python

import sys

seq = sys.stdin.readline().strip()

def complement(s):
  """Return the complementary sequence string."""
  basecomplement = {'A': 'T', 'C': 'G', 'G': 'C', 'T': 'A'}
  letters = list(s)
  letters = [basecomplement[base] for base in letters]
  return ''.join(letters)

print(complement(seq[::-1]))
